packages/plugin-core/src/components/views/PreviewLinkHandler.ts

```diff
@@ -50,103 +58,96 @@ export class PreviewLinkHandler implements IPreviewLinkHandler {
     data,
   }: {
     data: { id?: string | undefined; href?: string | undefined };
-  }): Promise<void> {
-    const linkType = this.classifyLink(data);
+  }) {
+    const ctx = "PreviewLinkHandler.onLinkClicked";
+    // If href is missing, something is wrong with our link handler. Just let the VSCode's default handle it.
+    if (!data.href) return LinkType.UNKNOWN;
+    // First check if it's a web URL.
+    if (isWebUri(data.href)) {
+      // There's nothing to do then, the default handler opens them automatically.
+      // If we try to open it too, it will open twice.
+      return LinkType.WEBSITE;
+    }
 
-    switch (linkType) {
-      case LinkType.ASSET: {
-        await this.handleAssetLink({
-          data,
-          wsRoot: this._ext.getDWorkspace().wsRoot,
+    const uri = vscode.Uri.parse(data.href);
+    // First, check if the URL matches any note
+    try {
+      const noteData = await this.getNavigationTargetNoteForWikiLink({
+        data,
+        engine: this._ext.getEngine(),
+      });
+
+      if (noteData.note) {
+        // Found a note, open that
+        await this._ext.commandFactory.goToNoteCmd().execute({
+          qs: noteData.note.fname,
+          vault: noteData.note.vault,
+          // Avoid replacing the preview
+          column: vscode.ViewColumn.One,
+          anchor: noteData.anchor,
         });
-        break;
+        return LinkType.WIKI;
       }
-      case LinkType.WIKI: {
-        try {
-          const noteData = await this.getNavigationTargetNoteForWikiLink({
-            data,
-            notes: this._ext.getEngine().notes,
-          });
-
-          if (noteData.note === undefined) {
-            // One valid case for note being undefined if user clicked on target note in
-            // different vault which had ambiguous vault and upon being prompted
-            // to select a note in quick user cancelled.
-            return;
-          }
-
-          this._ext.commandFactory.goToNoteCmd().execute({
-            qs: noteData.note.fname,
-            vault: noteData.note.vault,
+    } catch (err) {
+      Logger.error({ ctx, error: ErrorFactory.wrapIfNeeded(err) });
```

Comment:
`getNavigationTargetNoteForWikiLink` can throw if it fails to read the note id from the link.

I think `GotoNote` can also throw if the note doesn't exist, because it grabs a note from ID then casts it:
```ts
const note = data?.note as NoteProps;
```

I could refactor both functions to avoid throwing, but I also like having the try block as a fallback. Attempting to process it as an asset is a safe operation because we first check if the file actually exists.

Reply:
I guess to me - either we have an expected error (expected meaning valid code logic but maybe invalid user behavior), in which case it makes to continue, or else it's an unexpected error and we should stop further execution in the function at this point. 

If it's expected, then let's not log it as `Logger.error` (maybe Logger.debug), as this will cause uploads to Sentry, and I'd rather err on less data than more data right now because it's still too noisy.